### PR TITLE
Bug 568152 - Use FrameworkUtil#getBundle instead of Platform#getBundle

### PR DESCRIPTION
**Pull request by kthoms**
From a performance view it is recommended to use
FrameworkUtil#getBundle() instead of Platform.getBundle() when a class
from the referred bunlde is accessible.

This change replaces the usage of the method for the generated
Executable Extension Factory.

Note: This is the generator-only change. Changes to generated code will come by separate PR.

### CODE REVIEW

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend [hunk 1]
```diff
@@ -466,7 +466,7 @@ class XtextGeneratorTemplates {
 			
 				@Override
 				protected �'org.osgi.framework.Bundle'.typeRef� getBundle() {
-					return �'org.eclipse.core.runtime.Platform'.typeRef�.getBundle(�eclipsePluginActivator�.PLUGIN_ID);
+					return �'org.osgi.framework.FrameworkUtil'.typeRef�.getBundle(�eclipsePluginActivator�.class);
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend:469]: is this on classpath? can we use real typeref?

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend [hunk 2]
```diff
@@ -466,7 +466,7 @@ class XtextGeneratorTemplates {
 			
 				@Override
 				protected �'org.osgi.framework.Bundle'.typeRef� getBundle() {
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend:468]: same here

**Comment by cdietrich** [comment]
can you please provide follow up PRs regerating all production and test languages

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend (hunk 1 above)
Comment (kthoms) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend:469]: yes
Bildschirmfoto 2020-10-26 um 08 51 50

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend (hunk 2 above)
Comment (kthoms) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend:468]: ok, will change that, too.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend:469]: I think the point was to use the real class in the template.

**Comment by szarnekow** [comment]
> can you please provide follow up PRs regerating all production and test languages

Please create the follow-up merge requests next to this one such that we can see whether the build is green before we merge this change.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend:469]: yes

Comment (cdietrich) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend:469]: org.osgi.framework.FrameworkUtil.typeRef with no ''

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend (hunk 2 above)
Comment (kthoms) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend:468]: What means 'same here'? The method signature does not change.

**Comment by kthoms** [comment]
> > can you please provide follow up PRs regerating all production and test languages
> 
> Please create the follow-up merge requests next to this one such that we can see whether the build is green before we merge this change.

Will do.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend (hunk 2 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.xtend:468]: use class instead of string

**Comment by cdietrich** [comment]
there is also a handfull of places where we use this in code

File: org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.java [hunk 3]
```diff
@@ -47,6 +47,8 @@
 import org.eclipse.xtext.xtext.generator.model.annotations.IClassAnnotation;
 import org.eclipse.xtext.xtext.generator.model.annotations.SuppressWarningsAnnotation;
 import org.eclipse.xtext.xtext.generator.model.project.IXtextProjectConfig;
+import org.osgi.framework.Bundle;
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.java:50]: should we also check others in the class like org.osgi.framework.BundleContext

Comment (cdietrich) [reply org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.java:50]: ping @kthoms

Comment (kthoms) [reply org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.java:50]: OK, I'll update this.

Comment (kthoms) [reply org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/XtextGeneratorTemplates.java:52]: done

**Review by cdietrich — APPROVED** [review]
lets wait for a GREEN BUILD

**Pull request merged**